qiskit/circuit/quantumcircuit.py

```diff
@@ -1079,7 +1107,32 @@ def __init__(
             regs = tuple(int(reg) for reg in regs)  # cast to int
         self._base_name = None
         self.name: str
-        """A human-readable name for the circuit."""
+        """A human-readable name for the circuit.
+        
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+                :context: reset
+
+                from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+
+                qr = QuantumRegister(2)
+                cr = ClassicalRegister(2)
+                qc = QuantumCircuit(qr, cr)
+
+                qc.h(qr[0])
+                qc.cx(qr[0], qr[1])
+                qc.measure(qr, cr)
+                qc.name = "my_circuit"
+
+                print(qc.name)
```

Comment:
I recommend removing as much as possible that isn't related to the method/attribute. For example, we can remove the registers and operations. In this case, it might also be helpful to show you can set the name through the constructor.

```suggestion
                from qiskit import QuantumCircuit

                qc = QuantumCircuit(2, 2, name="my_circuit")
                print(qc.name)
```